src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c

```diff
@@ -6088,15 +6088,17 @@ int wm_vuldet_remove_table(sqlite3 *db, char *table) {
 
 int wm_vuldet_index_nvd(sqlite3 *db, update_node *upd, nvd_vulnerability *nvd_it) {
     time_t index_time = time(NULL);
-    int result;
+    int result = 0;
     int cve_count = 0;
 
     // Clean everything only if there are still residues of an offline update
+    // made by multi_path update. For the cases of multi_url or the online
+    // update, we just need to clean the NVD for the year being indexed.
     if (result = wm_vuldet_has_offline_update(db), result == OS_INVALID) {
         goto error;
-    } else if (result == 1 && wm_vuldet_clean_nvd(db)) {
+    } else if (result == 1 && upd->multi_path && wm_vuldet_clean_nvd(db)) {
```

Comment:
The call to this function in the `multi_url` update was causing as a result a database with only the CVEs of the last year being indexed.
Considering that in the `multi_url` update the behavior is the same that the online update, I don't see any reason for what we shouldn't follow the same behavior and only remove the CVEs of the year being indexed in the next `else if` statement.